remsfal-service/src/test/java/de/remsfal/service/control/GarageControllerTest.java

```diff
@@ -0,0 +1,170 @@
+package de.remsfal.service.control;
+
+import de.remsfal.core.json.project.GarageJson;
+import de.remsfal.core.json.project.ImmutableGarageJson;
+import de.remsfal.core.model.project.BuildingModel;
+import de.remsfal.core.model.project.GarageModel;
+import de.remsfal.core.model.project.PropertyModel;
+import de.remsfal.service.AbstractTest;
+import de.remsfal.service.TestData;
+import de.remsfal.service.entity.dto.GarageEntity;
+import io.quarkus.test.junit.QuarkusTest;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.NotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+@QuarkusTest
+class GarageControllerTest extends AbstractTest {
+
+    @Inject
+    PropertyController propertyController;
+
+    @Inject
+    BuildingController buildingController;
+
+    @Inject
+    GarageController garageController;
+
+    @BeforeEach
+    void setupTestProjects() {
+        runInTransaction(() -> entityManager
+                .createNativeQuery("INSERT INTO PROJECT (ID, TITLE) VALUES (?,?)")
+                .setParameter(1, TestData.PROJECT_ID)
+                .setParameter(2, TestData.PROJECT_TITLE)
+                .executeUpdate());
+    }
+
+    @Test
+    void createGarage_SUCCESS_idGenerated() {
+        final PropertyModel property = propertyController.createProperty(TestData.PROJECT_ID, TestData.propertyBuilder().build());
+        assertNotNull(property.getId(), "Property ID should not be null");
+
+        final BuildingModel building = TestData.buildingBuilder()
```

Comment:
See above: A lot of repeated steps for each test method. Consider extracting common setup logic into helper method, to reduce code redundancy and shorten the test class